envoy/common/exception.h

```diff
@@ -63,8 +63,8 @@ class EnvoyException : public std::runtime_error {
   }
 
 #define RETURN_IF_NOT_OK(variable)                                                                 \
-  if (!variable.ok()) {                                                                            \
-    return variable;                                                                               \
+  if (absl::Status temp_status = variable; !temp_status.ok()) {                                    \
+    return temp_status;                                                                            \
```

Comment:
Could you explain or comment why?

Reply:
commented inline but one of the tests failed because I was doing 
RETURN_IF_NOT_OK(function) and it was doing
if (!function.ok()) {
  return function
}
and failing because mocks only expected function to be called once.  Added the temp variable both for tests and to minimize work.